### PR TITLE
Remove requestPermission().

### PR DESCRIPTION
**Pull request by rsolomakhin**
@romandev and @marcoscaceres: Please take a look. This is a follow up to my issue https://github.com/w3c/payment-handler/issues/246.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -554,13 +520,9 @@ <h2>
...
           </p>
           <ol>
-            <li>Let <var>permission</var> be the result of running
-              <a data-cite="!permissions#dfn-retrieve-the-permission-state">retrieving
-              the permission state</a> of the permission associated with
-              <a>payment handler</a>'s <a>origin</a>.
-            </li>
-            <li>If <var>permission</var> is not "granted", then return a
-            <a>Promise</a> rejected with a <a>NotAllowedError</a>.
+            <li>Upon user agent discretion and depending on user consent,
```
Comment (marcoscaceres) [inline index.html]: This feels like a side-effect, but maybe it isn't. I guess we need to describe this further as to how user consent is acquired.

**Comment by rsolomakhin** [comment]
@marcoscaceres: I've described further how user consent is acquired in the notes following the `set()` algorithm. Please take a look.

File: index.html [hunk 2]
```diff
@@ -615,6 +577,32 @@ <h2>
             <li>Resolve <var>p</var>.
             </li>
           </ol>
+          <p class="note" title="Payment Handler Install">
+            Calling <a>set()</a> for the first time is equivalent to installing
+            a payment handler for this site.
+          </p>
```
Comment (ianbjacobs) [inline index.html]: @rsolomakhin, I'm not sure what that statement means or its impact. Could you explain a bit more?

File: index.html [hunk 3]
```diff
@@ -615,6 +577,32 @@ <h2>
...
+            following, for example:
+            <ul>
+              <li>Accepting or rejecting a permission prompt dialog at the time
+              of <a>set()</a> call</li>
+              <li>A user agent setting to always allow or deny <a>set()</a>
+              calls from specific origins</li>
+              <li>A user agent setting to always allow or deny all calls to
+              <a>set()</a></li>
+            </ul>
+          </p>
```
Comment (ianbjacobs) [inline index.html]: @rsolomakhin, as I read the above note, it feels to me like it should be moved to the section on security and privacy considerations and rewritten something like this:

<p class='note'>
   This specification does not define how the user agent establishes user consent for <a>set()</a>. The user agent might prompt the user as part of the <a>set()</a> call, or allow the user manual configuration options for specific origins or all origins.
</p>

File: index.html [hunk 4]
```diff
@@ -615,6 +577,32 @@ <h2>
...
+          <p class="note" title="User Agent Discretion">
+            A user agent may reject or accept the <a>set()</a> promise without
+            prompting the user for consent using any of the following
+            heuristics, for example:
+            <ul>
+              <li>Site engagement metrics</li>
+              <li>Safe browsing database</li>
+              <li>SSL certificate validity</li>
+            </ul>
+          </p>
```
Comment (ianbjacobs) [inline index.html]: @rsolomakhin, I don't yet understand this part of the proposal. It seems that partly you would like to authorize browsers to do some things without user consent when bad things happen, such as detecting a malicious site or bad certificate. If that's the case, could we rephrase this as "may reject the set() promise for security reasons" (and then list some examples)?

However, I am not understanding the use cases for accepting the set() promise without user consent.

**Review by ianbjacobs — COMMENTED** [review]
Hi @rsolomakhin, see my inline comments.

File: index.html (hunk 2 above)
Comment (rsolomakhin) [reply index.html]: Sure, @ianbjacobs. Chrome will persist a list `L` of origins that are payment handlers. When an origin calls `instruments.set()`, Chrome will check whether this origin is in the list `L`.

If the origin is in the list `L`, then proceed with the rest of the algorithm, because this is an update of instruments in an existing payment handler.

If the origin is _not_ in the list `L`, then use heuristics to determine whether to deny the `.set()` promise outright (for bad actors, for example), allow the `.set()` algorithm to proceed silently (due to user settings, for example), or prompt the user via a dialog with a question like "Install payment app from origin X?" If the `.set()` algorithm runs to successful completion, then add the origin to `L`.

File: index.html (hunk 4 above)
Comment (rsolomakhin) [reply index.html]: > ot understanding the use cases for accepting the set() promise without user consent.

One use case is high site engagement stat for this user on this origin.

**Comment by rsolomakhin** [comment]
Hi @ianbjacobs,

Please see my comments inline. If I've answered your questions to satisfaction, then I will make the edits you have requested for rephrasing and re-arranging the pull request.

Cheers,
Rouslan

File: index.html (hunk 2 above)
Comment (ianbjacobs) [reply index.html]: Hi @rsolomakhin,

If I understand, the way that one registers a handler is now through set(). If that's the case, then I suggest as an alternative to your note, the following text at the very beginning of section 3. (Registration):

 <p>One registers a payment handler with the user agent when assigning the first <a>PaymentInstrument</a> to it through the <a>set()</a> method.</p>

File: index.html (hunk 3 above)
Comment (ianbjacobs) [reply index.html]: Here's an alternative (in light of other parts of this thread):

<p>This specification does not define how the user agent establishes user consent when <a>set()</a> is first called. The user agent might prompt the user as a result of <a>set()</a>, or might not if consent has been established previously through manual configuration by the user or usage patterns.</p>

<p>User agents MAY reject a <a>set()</a> promise for security reasons (e.g., due to an invalid SSL certificate) and SHOULD notify the user when this happens.</p>

How does that sound?

Ian

File: index.html (hunk 4 above)
Comment (ianbjacobs) [reply index.html]: Thanks, Rouslan. Let me know if the above text would work for you.

We will have to decide on whether to use MAY/SHOULD in this security considerations text.

Ian

**Comment by rsolomakhin** [comment]
To be continued in https://github.com/w3c/payment-handler/pull/277
